### PR TITLE
Better plan names

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes CartoDB/cartodb-central/issues/1518

@gfiorav CR this, please.

### CODE REVIEW

File: app/helpers/account_type_helper.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module AccountTypeHelper
+  # Customer-facing plan name. Front is responsible of shortening long ones.
+  def plan_name(account_type)
+    PLAN_NAME_BY_ACCOUNT_TYPE.fetch(account_type.downcase, account_type)
+  end
+
+  private
```
Comment (gfiorav) [inline app/helpers/account_type_helper.rb]: Curiosity: isn't this true for modules?

File: lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown_view.js [hunk 2]
```diff
@@ -30,16 +30,17 @@ module.exports = cdb.admin.DropdownMenu.extend({
...
-                        .replace(/CARTO /, '')
+                        .replace(/ - Monthly/i, ' - M')
+                        .replace(/ - Annual/i, ' - A')
+                        .replace(/Non-Profit/i, 'NP')
+                        .replace(/On-premises/i, 'OP')
+                        .replace(/Internal use engine/i, 'engine')
+                        .replace(/Lite/i, 'L')
+                        .replace(/CARTO for /i, '')
+                        .replace(/CARTO /i, '')
                         .toLowerCase();
```
Comment (gfiorav) [inline lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown_view.js]: Doesn't it make more sense to substitute for lowercase instead of for uppercase and then lowercasing again? i.e.: `.replace(/lump-sum/gi, 'lp')` instead of `.replace(/lump-sum/gi, 'LP')`

**Comment by juanignaciosl** [comment]
@gfiorav next CR, please. Many things have changed!

File: lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown_view.js [hunk 3]
```diff
@@ -22,6 +22,34 @@ module.exports = cdb.admin.DropdownMenu.extend({
...
+                        .replace(/ - Annual/i, ' - A')
+                        .replace(/Non-Profit/i, 'NP')
+                        .replace(/On-premises/i, 'OP')
+                        .replace(/Internal use engine/i, 'engine')
+                        .replace(/Lite/i, 'L')
+                        .replace(/Cloud Engine &/i, 'C. Engine &')
+                        .replace(/& Enterprise Builder/i, '& E. Builder')
+                        .replace(/CARTO for /i, '')
+                        .replace(/CARTO /i, '')
+                        .toLowerCase();
```
Comment (gfiorav) [inline lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown_view.js]: Still not sold no the double `.toLowerCase()`

File: app/helpers/account_type_helper.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    'CARTO for students - Annual' => 'CARTO for students',
+    'AMBASSADOR' => 'CARTO for Ambassadors',
+    'ire' => 'CARTO for Ambassadors',
+    'CARTO for Community - Annual' => 'CARTO for Community',
+    'Free Basemap LDS - Annual' => 'Free Basemap LDS',
+    'Enterprise LDS - Annual' => 'Enterprise LDS',
+    'CARTO Trial Account - Annual' => 'CARTO Trial Account'
+  }.freeze
+
+  PLAN_NAME_BY_ACCOUNT_TYPE_DOWN = Hash[PLAN_NAME_BY_ACCOUNT_TYPE.map { |k,v| [k.downcase, v] }]
```
Comment (gfiorav) [inline app/helpers/account_type_helper.rb]: Agree so if you have to make another commit for what ever reason it might be worth it to change this.

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cartodb-ui",
-  "version": "4.3.59",
+  "version": "4.3.59-ded06",
```
Comment (gfiorav) [inline package.json]: Forgot to say: I know this is so it would deploy in staging, but now checkout from master and bump after merge is the best way to avoid conflicts 👍 In case you didn't know!

File: lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown_view.js (hunk 3 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb/common/views/dashboard_header/settings_dropdown_view.js]: Sorry, I forgot your previous concern. What's your proposal? The final lowercasing is actually not needed, as the css uppercases it, so I'll remove it. Is it good enough?

File: package.json (hunk 5 above)
Comment (juanignaciosl) [inline package.json]: Yep, no problem. I labeled this PR with "do not merge yet" because of this.

Comment (gfiorav) [inline package.json]: I just like sharing my frontendy cool (not cool) tricks 😎

**Comment by juanignaciosl** [comment]
@gfiorav next CR, please

**Comment by juanignaciosl** [comment]
Not yet ;-)

File: app/helpers/account_type_helper.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
+# This file should be synched with Central
+module AccountTypeHelper
+  # Customer-facing plan name. Front is responsible of shortening long ones.
+  def plan_name(account_type)
+    PLAN_NAME_BY_ACCOUNT_TYPE_DOWN.fetch(account_type.downcase, account_type)
+  end
+
+  def plan_name_no_time(account_type)
```
Comment (javitonino) [inline app/helpers/account_type_helper.rb]: This is here just to keep in sync with Central, right? It's not actually used. Not opposed to have it duplicated, I think it's worth it for ease of sync. Just confirming.

Comment (juanignaciosl) [inline app/helpers/account_type_helper.rb]: Yes :-)
